Update Expression Language documentation with allowed group rules expressions (#3558)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/okta-expression-language/index.md

To include a granted scope array and convert it to a space-delimited string, use the following expression:<br>
`String.replace(Arrays.toCsvString(access.scope),","," ")`

## Expressions in group rules

> The set of expressions allowed in group rules is limited to `String`, `Arrays`, and `user` expressions.

Reviewer: I think we can be a bit more concise. May I suggest:
```suggestion
Group rules only allow `string`, `array`, and `user` expressions.
```
Also, I'm pretty sure these expression types don't need to be capitalized. We should have Susan Harper review this PR, too. She's our copyeditor and style guide guru.

Author: Updated - let me know what you think.

Author: I forgot to mention that `String` and `Arrays` refer to product functionality. Someone has to type in the capitalized version in group rules and can't use the lowercase version.

> For example, `user.hasBadge`, `String.stringContains(user.email, "@example.com")` and

Reviewer: I think these might be best presented as lists. It also makes the parallelism a bit easier to see:
```suggestion
The following examples are allowed in group rule expressions:

* `user.hasBadge`
* `String.stringContains(user.email, "@example.com")`
* `Arrays.contains(user.favoriteColors, "blue")`

The following example isn't allowed in group rule expressions:

`Convert.toInt("2018") == user.yearJoined`
```

Author: Updated - let me know what you think.

To include a granted scope array and convert it to a space-delimited string, use the following expression:<br>
`String.replace(Arrays.toCsvString(access.scope),","," ")`

## Expressions in group rules

Group rule conditions only allow `String`, `Arrays`, and `user` expressions.

For example, the following expressions are allowed in group rule conditions:

> * `user.hasBadge`

Reviewer 2: I need to check other doc but do we need to specify `hasBadge` is one of the custom property defined in schema? Same thing for `favoriteColors` is array property in custom schema?

Author: I will let @brentschaus-okta make the call.

Reviewer 3: I think it might be a good idea to use properties in this table and mention that these expression examples are based on the assumption that the user profile contains the properties mentioned in that table(?)

Author: I made an update.

To include a granted scope array and convert it to a space-delimited string, use the following expression:<br>
`String.replace(Arrays.toCsvString(access.scope),","," ")`

## Expressions in group rules

Group rule conditions only allow `String`, `Arrays`, and `user` expressions.

For example, given the user profile has a base string attribute called `email`, and assuming the user profile has a
custom boolean attribute called `hasBadge` and a custom string attribute called `favoriteColor`, the following
expressions are allowed in group rule conditions:

* `user.hasBadge`
* `String.stringContains(user.email, "@example.com")`
* `Arrays.contains(user.favoriteColors, "blue")`

> The following expression is not allowed in group rule conditions, even if the user profile has a custom integer

Reviewer 4: ```suggestion
The following expression isn't allowed in group rule conditions, even if the user profile has a custom integer
```
